### PR TITLE
feat: ledger stax + flex support

### PR DESCRIPTION
**Pull request by SWvheerden**
Description
---
Upgrading ledger library for stax and flex support

### CODE REVIEW

File: applications/minotari_ledger_wallet/wallet/src/app_ui/menu.rs [hunk 1]
```diff
@@ -45,3 +51,14 @@ pub fn ui_menu_main(comm: &mut Comm) -> Event<Instruction> {
         }
     }
 }
+
+#[cfg(any(target_os = "stax", target_os = "flex"))]
+pub fn ui_menu_main(_: &mut Comm) -> NbglHomeAndSettings {
+    // Load glyph from 64x64 4bpp gif file with include_gif macro. Creates an NBGL compatible glyph.
+    const FERRIS: NbglGlyph = NbglGlyph::from_include(include_gif!("key_64x64.gif", NBGL));
```
Comment (brianp) [inline applications/minotari_ledger_wallet/wallet/src/app_ui/menu.rs]:
```suggestion
    const TARI: NbglGlyph = NbglGlyph::from_include(include_gif!("key_64x64.gif", NBGL));
```

File: applications/minotari_ledger_wallet/wallet/src/app_ui/menu.rs [hunk 2]
```diff
@@ -45,3 +51,14 @@ pub fn ui_menu_main(comm: &mut Comm) -> Event<Instruction> {
...
+
+    // Display the home screen.
+    NbglHomeAndSettings::new()
+        .glyph(&FERRIS)
```
Comment (brianp) [inline applications/minotari_ledger_wallet/wallet/src/app_ui/menu.rs]:
```suggestion
        .glyph(&TARI)
```

Or anything that is more representative than `FERRIS` :D

File: applications/minotari_ledger_wallet/wallet/src/handlers/get_one_sided_metadata_signature.rs [hunk 3]
```diff
@@ -95,17 +107,35 @@ pub fn handler_get_one_sided_metadata_signature(comm: &mut Comm) -> Result<(), A
...
+            Some(&CROSSMARK),
+        );
+        if !review.show() {
+            return Err(AppSW::UserCancelled);
+        }
+    }
+    #[cfg(any(target_os = "stax", target_os = "flex"))]
+    {
+        // Load glyph from 64x64 4bpp gif file with include_gif macro. Creates an NBGL compatible glyph.
+        const FERRIS: NbglGlyph = NbglGlyph::from_include(include_gif!("key_64x64.gif", NBGL));
```
Comment (brianp) [inline applications/minotari_ledger_wallet/wallet/src/handlers/get_one_sided_metadata_signature.rs]: Same with `FERRIS`

What's an NBGL campatible glyph anyway? What's NBGL?

File: applications/minotari_ledger_wallet/wallet/src/handlers/get_script_offset.rs [hunk 4]
```diff
@@ -199,6 +202,7 @@ pub fn handler_get_script_offset(
     comm.append(&[RESPONSE_VERSION]); // version
     comm.append(&script_offset.to_vec());
     offset_ctx.reset();
+    offset_ctx.finished = true;
```
Comment (brianp) [inline applications/minotari_ledger_wallet/wallet/src/handlers/get_script_offset.rs]: Was this to do with your state issue?
What exactly is the bool solving, I think I'm overlooking its purpose.

File: applications/minotari_ledger_wallet/wallet/src/main.rs [hunk 5]
```diff
@@ -40,62 +36,27 @@ use handlers::{
...
-
-unsafe impl critical_section::Impl for MyCriticalSection {
-    unsafe fn acquire() -> RawRestoreState {
-        // nothing, it's all good, don't worry bout it
-    }
-
-    unsafe fn release(_token: RawRestoreState) {
-        // nothing, it's all good, don't worry bout it
-    }
-}
```
Comment (brianp) [inline applications/minotari_ledger_wallet/wallet/src/main.rs:94]: Was this tested on the nanos+ ? 

I was under the distinct impression without defining the allocator we lose the ability to well... Allocate anything. Essentially it's memory-less

**Review by brianp — COMMENTED** [review]
Some changes, and questions.

I don't love that this is how they want us to write apps. I'd almost be more inclined to write two different ones, and modularize all the functional bits so we just wrap it in two different UX's. 

*but* the work is done now, and it is how they suggest doing it so 🤷🏻

File: applications/minotari_ledger_wallet/wallet/src/main.rs (hunk 5 above)
Comment (SWvheerden) [reply applications/minotari_ledger_wallet/wallet/src/main.rs:94]: yes it was, 
Ledger realised that they need one, so the rust_sdk lib now includes one and it clashes as you can only have one defined.

File: applications/minotari_ledger_wallet/wallet/src/handlers/get_script_offset.rs (hunk 4 above)
Comment (SWvheerden) [reply applications/minotari_ledger_wallet/wallet/src/handlers/get_script_offset.rs]: an attempt at getting the UI working, good catch this needs to be removed

File: applications/minotari_ledger_wallet/wallet/src/main.rs (hunk 5 above)
Comment (brianp) [reply applications/minotari_ledger_wallet/wallet/src/main.rs:94]: Ahhh neat.

**Review by brianp — APPROVED** [review]
Ack

**Pull request merged**